File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for(int i=0;i<commands.size();i++)
    {
        int start = commands[i][0];
        int end = commands[i][1];
        int idx = commands[i][2];
        
        vector<int> nums;
        for(int j=start-1;j<end;j++)
        {
            nums.push_back(array[j]);
        }
        
        sort(nums.begin(), nums.end());
        
        answer.push_back(nums[idx-1]);
    }
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        
        int[] answer = new int[commands.length];
        
        for(int i=0;i<commands.length;i++) {
            int s = commands[i][0]-1;
            int e = commands[i][1]-1;
            int p = commands[i][2]-1;
            
            int[] nums = new int[e-s+1];
            int t=0;
            for(int j=s;j<=e;j++){
                nums[t++]=array[j];
            }
            Arrays.sort(nums);
            answer[i]=nums[p];
        }
        return answer;
    }
}